Translation of blog post: "2016-09-28-our-first-50000-stars" (#242)
reactjs/pt-BR.reactjs.org: content/blog/2016-09-28-our-first-50000-stars.md

---

> title: "Nossas primeiras 50,000 stars no GitHub"

Reviewer: ```suggestion
title: "Nossas primeiras 50.000 stars no GitHub"
```

author: [vjeux]
---

> Apenas três anos e meio após nós criarmos uma pequena biblioteca Javascript chamada React. A jornada desde aquele dia tem sido incrivelmente excitante.

Reviewer: Acredito que o sentido da primeira frase está errado.
É algo no sentido de: "Há apenas três anos e meio nós abrimos o código de uma pequena biblioteca chamada React."

Na segunda frase faz sentido uso "desde esse dia" porque acaba de ser citado na frase anterior do mesmo parágrafo.

> Apenas três anos e meio após nós criarmos uma pequena biblioteca Javascript chamada React. A jornada desde aquele dia tem sido incrivelmente excitante.

Reviewer: ```suggestion
Há apenas três anos e meio nós abrimos o código de uma pequena biblioteca chamada React. A jornada desde este dia tem sido incrivelmente excitante.
```

## Camiseta Comemorativa {#commemorative-t-shirt}

> Para comemorar as 50,000 stars no GitHub, [Maggie Appleton](http://www.maggieappleton.com/) do [egghead.io](http://egghead.io/) desenhou para nós uma camiseta especial, que está disponível para compra no Teespring **somente por uma semana** até quinta feira, 06 de Outubro. Maggie também escreveu [um post](https://www.behance.net/gallery/43269677/Reacts-50000-Stars-Shirt) mostrando todos os diferentes conceitos que ela criou antes de escolher como seria o design final.

Reviewer: ```suggestion
Para comemorar as 50.000 stars no GitHub, [Maggie Appleton](http://www.maggieappleton.com/) do [egghead.io](http://egghead.io/) desenhou para nós uma camiseta especial, que está disponível para compra no Teespring **somente por uma semana** até quinta feira, 06 de Outubro. Maggie também escreveu [um post](https://www.behance.net/gallery/43269677/Reacts-50000-Stars-Shirt) mostrando todos os diferentes conceitos que ela criou antes de escolher como seria o design final.
```

Eu traduziria estrelas também.

Pessoalmente, gosto de traduzir post para publicação.

<a target="_blank" href="https://teespring.com/react-50000-stars"><img src="../images/blog/react-50k-tshirt.jpg" width="650" height="340" alt="React 50k Tshirt" /></a>

> As camisetas são super macias usando o tecido tri-blend da American Apparel's. Nós também temos para crianças, camisetas infantis e macacões para bebês disponíveis.

Reviewer: ```suggestion
As camisetas são supermacias usando o tecido tri-blend da American Apparel's. Nós também temos para crianças, camisetas infantis e macacões para bebês disponíveis.
```

Acredito que o sentido da segunda frase seja mais próximo de: "Nós também temos camisetas para crianças e macacões ..."

Pesquisando, onesie parece uma peça de roupa específica.

* [Camisetas para Adultos (corte reto e ajustadas)](https://teespring.com/react-50000-stars)
* [Camisetas para crianças](https://teespring.com/react-50000-stars-kids)
* [Camisetas Infantis](https://teespring.com/react-50000-stars-toddler)

> * [Camisetas para bêbês](https://teespring.com/react-50000-stars-baby)

Reviewer: Onesies não são camisetas

> * [Camisetas para bêbês](https://teespring.com/react-50000-stars-baby)

Reviewer: ```suggestion
* [Onesies para bêbês](https://teespring.com/react-50000-stars-baby)
```

> Todos os ganhos das camisetas serão doados para o [CODE2040](http://www.code2040.org/), uma organização sem fins lucrativos que cria acesso, conscientização e oportunidades em tecnologia para minorias não representadas com um foco especifico em talentos afrodescendente e latinos.

Reviewer: No sentido mais literal ficaria: "Proventos das camisetas serão doados".

> Todos os ganhos das camisetas serão doados para o [CODE2040](http://www.code2040.org/), uma organização sem fins lucrativos que cria acesso, conscientização e oportunidades em tecnologia para minorias não representadas com um foco especifico em talentos afrodescendente e latinos.

Reviewer: ```suggestion
Todos os proventos das camisetas serão doados para o [CODE2040](http://www.code2040.org/), uma organização sem fins lucrativos que cria acesso, conscientização e oportunidades em tecnologia para minorias não representadas com um foco especifico em talentos afrodescendente e latinos.
```

## Arqueologia {#archeology}

> Nós gastamos muito tempo tentando explicar conceitos que existem atras do React e problemas que ele busca resolver, mas nós não temos falamos muito sobre como o React evoluiu antes de se tornar um projeto de código aberto. Este momento parece bom para lembrarmos os primeiros commits e compartilharmos alguns dos momentos importantes e fatos engraçados.

Reviewer 2: ```suggestion
Gastamos muito tempo tentando explicar conceitos que existem atrás do React e problemas que ele busca resolver, mas não falamos muito sobre como o React evoluiu antes de se tornar um projeto de código aberto. Este momento parece bom para lembrarmos os primeiros commits e compartilharmos alguns dos momentos importantes e fatos engraçados.
```

Author: Ajustado. Obrigado pelo comentário!

//  <div> wheels</div>
//</div>
```

> Bolt introduziu um número de APIs e recursos que acabariam eventualmente entrando no React, incluindo o método `render`, `createClass`, e `refs`. Bolt introduziu o conceito de `refs` como uma forma de criar referências para nós que podem ser utilizados imperativamente. Está maneira foi relevante para a interoperabilidade herdada e a adoção incremental, e enquanto o React se esforçaria para ser muito mais funcional, `refs` provou ser uma forma útil de sair do paradigma funcional quando surgir a necessidade.

Reviewer: Está faltando uma quebra de linha antes deste parágrafo.

> Bolt introduziu um número de APIs e recursos que acabariam eventualmente entrando no React, incluindo o método `render`, `createClass`, e `refs`. Bolt introduziu o conceito de `refs` como uma forma de criar referências para nós que podem ser utilizados imperativamente. Está maneira foi relevante para a interoperabilidade herdada e a adoção incremental, e enquanto o React se esforçaria para ser muito mais funcional, `refs` provou ser uma forma útil de sair do paradigma funcional quando surgir a necessidade.

Reviewer: Não sei se há a necessidade de explicitar "método".

```suggestion
Bolt introduziu um número de APIs e recursos que acabariam eventualmente entrando no React, incluindo o método `render`, `createClass`, e `refs`. Bolt introduziu o conceito de `refs` como uma forma de criar referências para nós que podem ser utilizados imperativamente. Esta maneira foi relevante para a interoperabilidade herdada e a adoção incremental, e enquanto o React se esforçaria para ser muito mais funcional, `refs` provou ser uma forma útil de sair do paradigma funcional quando surgir a necessidade.
```

Achei a última frase com sentido diferente da original.

> Bolt introduziu um número de APIs e recursos que acabariam eventualmente entrando no React, incluindo o método `render`, `createClass`, e `refs`. Bolt introduziu o conceito de `refs` como uma forma de criar referências para nós que podem ser utilizados imperativamente. Está maneira foi relevante para a interoperabilidade herdada e a adoção incremental, e enquanto o React se esforçaria para ser muito mais funcional, `refs` provou ser uma forma útil de sair do paradigma funcional quando surgir a necessidade.

Reviewer: ```suggestion

Bolt introduziu um número de APIs e recursos que acabariam eventualmente entrando no React, incluindo o método `render`, `createClass`, e `refs`. Bolt introduziu o conceito de `refs` como uma forma de criar referências para nós que podem ser utilizados imperativamente. Está maneira foi relevante para a interoperabilidade herdada e a adoção incremental, e enquanto o React se esforçaria para ser muito mais funcional, `refs` provou ser uma forma útil de sair do paradigma funcional quando surgir a necessidade.
```

//</div>
```

> Boltz introduziu um número de APIs e recursos que acabariam eventualmente entrando no React, incluindo o método `render`, `createClass`, e `refs`. Bolt introduziu o conceito de `refs` como uma forma de criar referências para nós que podem ser utilizados imperativamente. Está maneira foi relevante para a interoperabilidade herdada e a adoção incremental, e enquanto o React o React se esforça para ser tornar muito mais funcional, `refs` provou ser uma forma útil de sair do paradigma funcional quando surgir a necessidade.

Reviewer 2: ```suggestion
Bolt introduziu um número de APIs e recursos que acabariam eventualmente entrando no React, incluindo o método `render`, `createClass`, e `refs`. Bolt introduziu o conceito de `refs` como uma forma de criar referências para nós que podem ser utilizados imperativamente. Está maneira foi relevante para a interoperabilidade herdada e a adoção incremental, e enquanto o React se esforçaria para ser muito mais funcional, `refs` provou ser uma forma útil de sair do paradigma funcional quando surgir a necessidade.
```

Author: Ajustado. Obrigado.

};
```

> ## Nasce o FBolt {#fbolt-is-born}

Reviewer: Problemas com número de linhas corretos.

> Através do seu experimento FaxJS, Jordan  se convence que as APIs funcionais - que desencoraja a mutação - oferece uma melhor, mais escalável forma de construir interfaces de usuários. Ele importou sua biblioteca na base de códigos do Facebook em Março de 2012 e renomeou a biblioteca para "FBolt", significando uma extensão do Bolt onde os componentes são escritos em um estilo de programação funcional. Ou talvez "Fbolt" foi uma prévia do FaxJS - ele não nos contou! ;)

Reviewer: ```suggestion
Através do seu experimento FaxJS, Jordan se convence de que as APIs funcionais - que desencorajam mutação - oferecem uma melhor, mais escalável forma de construir interfaces de usuários. Ele importou sua biblioteca na base de códigos do Facebook em Março de 2012 e renomeou a biblioteca para "FBolt", significando uma extensão do Bolt onde os componentes são escritos em um estilo de programação funcional. Ou talvez "Fbolt" foi uma prévia do FaxJS - ele não nos contou! ;)
```

A interoperabilidade entre FBolt e Bolt permitiu experimentarmos a substituição de apenas um componente por vez com APIs de componentes mais funcionais. Poderíamos visitar as águas desse novo paradigma funcional, sem ter que entrar totalmente nele. Iniciamos com os componentes que estavam claramente melhor expressos funcionalmente e então continuamos depois com os demais componentes que poderíamos expressar como funções.

Percebendo que o FBolt não seria um ótimo nome para a biblioteca quando usada sozinha, Jordan Walke e [Tom Occhino](https://twitter.com/tomocchino) decidiram por um novo nome: "React." Depois de Tom ter renomeado a biblioteca inteira, Jordan comentou:

> Jordan Walke:

> Eu poderia acrescentar, que por uma questão de discussão, que muitos sistemas demonstram algum tipo de reatividade, mas geralmente exigem que você defina algum tipo de ouvinte (listeners) ponto a ponto e não funcionará com dados estruturados. Essa API reage a qualquer alteração de estado ou propriedade, e trabalha com dados de qualquer forma (tão profundamente quanto o próprio gráfico), então acho que o nome é adequado.

Reviewer: ```suggestion
Eu poderia acrescentar, que por uma questão de discussão, que muitos sistemas demonstram algum tipo de reatividade, mas geralmente exigem que você defina algum tipo de ouvinte (listeners) ponto a ponto e não funcionará com dados estruturados. Essa API reage a qualquer alteração de estado ou propriedade e trabalha com dados de qualquer forma (tão profundamente quanto o próprio gráfico), então acho que o nome é adequado.
```

};
```

## Nasce o FBolt {#fbolt-is-born}

> Através do seu experimento FaxJS, Jordan  se convence que as APIs funcionais - que desencoraja a mutação - oferece uma melhor, mais escalável forma de construir interfaces de usuários. Ele importou sua biblioteca na base de códigos do Facebook em Março de 2012 e renomeou a biblioteca para "FBolt", significando uma extensão do Bolt odne os componentes são escritos em um estilo de programação funcional. Ou talvez "Fbolt" foi uma prévia do FaxJS - ele não nos contou! ;)

Reviewer 2: ```suggestion
Através do seu experimento FaxJS, Jordan  se convence que as APIs funcionais - que desencoraja a mutação - oferece uma melhor, mais escalável forma de construir interfaces de usuários. Ele importou sua biblioteca na base de códigos do Facebook em Março de 2012 e renomeou a biblioteca para "FBolt", significando uma extensão do Bolt onde os componentes são escritos em um estilo de programação funcional. Ou talvez "Fbolt" foi uma prévia do FaxJS - ele não nos contou! ;)
```

Author: Ajustado.

> A interoperabilidade entre FBolt e Bolt permitiu a nós experimentar  a substituição de apenas um componente por vez com APIs de componentes mais funcionais. Nós poderíamos visitar as águas desse novo paradigma funcional, sem ter que entrar totalmente nele. Nós iniciamos com os componentes que estavam claramente melhor expressos funcionalmente e então continuamos depois com os demais componentes que poderíamos expressar como funções.

Reviewer 2: ```suggestion
A interoperabilidade entre FBolt e Bolt permitiu a experimentarmos a substituição de apenas um componente por vez com APIs de componentes mais funcionais. Poderíamos visitar as águas desse novo paradigma funcional, sem ter que entrar totalmente nele. Iniciamos com os componentes que estavam claramente melhor expressos funcionalmente e então continuamos depois com os demais componentes que poderíamos expressar como funções.
```

Author: Referente a remoção dos "nós" realmente, acho que ficou melhor.

Porém, na primeira linha, eu particulamente acho estranho a frase "A interoperabilidade entre FBolt e Bolt permitiu a experimentarmos a substituição". 

Acho que talvez deveria remover o "a" apos o permitiu, ficando:

"A interoperabilidade entre FBolt e Bolt permitiu experimentarmos a substituição".

Vou ajustar dessa última maneira.. qualquer coisa se não acharem correto, eu coloco o "a" novamente.

> Jordan Walke:
> Para a primeira etapa, eu proponho que nós façamos o mais fácil e mais genérico possível. Minha sugestão é simplesmente mapear expressões xml para funções de chamada.
>
> - `<x></x>` becomes `x( )`
> - `<x height=12></x>` becomes `x( {height:12} )`
> - `<x> <y> </y> </x>` becomes `x({ childList: [y( )] })`
>
> Neste ponto, o JSX não precisa conhecer o React - isso é apenas uma forma conveniente de escrever chamadas de função. Coincidentemente, a abstração primária do React é a função. Ok, talvez isso não seja uma coincidência ;)

Adam fez um comentário muito perspicaz, no qual agora é a forma padrão que nós escrevemos listas no React com JSX.

> Adam Hupp:
> Eu penso que nós deveríamos tratar matrizes de elementos como um frag. Isso é muito útil para construções como:
>
> ```js
> <ul>{foo.map(function(i) { return <li>{i.data}</li>; })}</ul>
> ```
>
> Neste caso, ul(...) irá receber um childList com um único filho, que é uma lista.

> React não utilizou a implementação de Adam diretamente. Ao invés, nós criamos JSX fazendo o fork do projeto [js-xml-literal](https://github.com/laverdet/js-xml-literal), um projeto paralelo do criado do XHP, Marcel Laverdet. O JSX tomou o nome de js-xml-literal, que Jordan modificou para ser apenas um acréscimo sintático para chamadas de funções profundamente aninhadas.

Reviewer 2: ```suggestion
React não utilizou a implementação de Adam diretamente. Ao invés, nós criamos JSX fazendo o fork do projeto [js-xml-literal](https://github.com/laverdet/js-xml-literal), um projeto paralelo do criador do XHP, Marcel Laverdet. O JSX tomou o nome de js-xml-literal, que Jordan modificou para ser apenas um acréscimo sintático para chamadas de funções profundamente aninhadas.
```

Author: Resolvido

## API Churn {#api-churn}

Durante o primeiro ano do React, as adoções internas estavam crescendo, mas havia muita rotatividade nas APIs dos componentes e nas convenções de nomenclatura:

* `project` foi renomeado para `declare`, depois para `structure` e finalmente para `render`.
* `Componentize` foi renomeado para `createComponent` e finalmente `createClass`.

Como o projeto estava prestes a se tornar aberto, [Lee Byron](https://twitter.com/leeb) sentou-se com Jordan Walke, Tom Occhino e Sebastian Markbåge com o objetivo de refatorar, reimplementar e renomear, um dos recursos mais amado do React - o ciclo de vida da API. Lee [criou uma API bem projetada](https://gist.github.com/vjeux/f2b015d230cc1ab18ed1df30550495ed) que ainda está sendo utilizada hoje em dia.

* Conceitos
    * component - Uma instância de ReactComponent
    * state - Estado interno de um componente
    * props - Estado externo de um componente
    * markup - A string/coisas HTML que o componente gera
    * DOM - O documento e elementos dentro do documento
* Ações
    * mount - Colocar um componente em um DOM
    * initialize - Preparar um componente para renderização

>     * update - Uma transiçaõ de um estado (e propriedades) resultando em renderização

Reviewer 2: ```suggestion
    * update - Uma transição de um estado (e propriedades) resultando em renderização
```
